src/DIRAC/WorkloadManagementSystem/DB/ElasticPilotParametersDB.py

```diff
@@ -0,0 +1,179 @@
+""" Module containing a front-end to the ElasticSearch-based ElasticPilotParametersDB.
+
+    The following class methods are provided for public usage
+      - getPilotParameters()
+      - setPilotParameter()
+      - deletePilotParameters()
+"""
+from DIRAC import S_OK, S_ERROR, gConfig
+from DIRAC.Core.Utilities import TimeUtilities
+from DIRAC.ConfigurationSystem.Client.PathFinder import getDatabaseSection
+from DIRAC.ConfigurationSystem.Client.Helpers import CSGlobals
+from DIRAC.Core.Base.ElasticDB import ElasticDB
+
+
+mapping = {
+    "properties": {
+        "PilotID": {"type": "long"},
+        "timestamp": {"type": "date"},
+        "CPUNormalizationFactor": {"type": "long"},
+        "Wallclock": {"type": "long"},
+        "HostName": {"type": "keyword"},
+        "GridCE": {"type": "keyword"},
+        "ModelName": {"type": "keyword"},
+        "Status": {"type": "keyword"},
+    }
+}
+
+
+class ElasticPilotParametersDB(ElasticDB):
+    def __init__(self, parentLogger=None):
+        """Standard Constructor"""
+
+        try:
+            section = getDatabaseSection("WorkloadManagement/ElasticPilotParametersDB")
+            indexPrefix = gConfig.getValue(f"{section}/IndexPrefix", CSGlobals.getSetup()).lower()
+
+            # Connecting to the ES cluster
+            super().__init__("WorkloadManagement/ElasticPilotParametersDB", indexPrefix, parentLogger=parentLogger)
+        except Exception as ex:
+            self.log.error("Can't connect to ElasticPilotParametersDB", repr(ex))
+            raise RuntimeError("Can't connect to ElasticPilotParametersDB") from ex
+
+        self.indexName_base = f"{self.getIndexPrefix()}_elasticpilotparameters_index"
+
+    def _indexName(self, pilotID: int) -> str:
+        """construct the index name
+
+        :param pilotID: Pilot ID
+        """
+        indexSplit = int(pilotID) // 1e6
```

Comment:
I guess you actually want this to be:
```suggestion
        indexSplit = int(pilotID // 1e6)
```